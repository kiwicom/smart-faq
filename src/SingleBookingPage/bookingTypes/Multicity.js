// @flow

import * as React from 'react';
import { createFragmentContainer, graphql } from 'react-relay';

import AccordionTitle from '../bookingItem/AccordionTitle';
import Accordion from '../bookingItem/Accordion';
import type { Multicity_booking } from './__generated__/Multicity_booking.graphql';

type Props = {|
  booking: Multicity_booking,
|};

const Multicity = ({ booking }: Props) => (
  <div>
    <AccordionTitle title="Trips" />
    {Array.isArray(booking.trips) &&
      booking.trips.map((trip, i) => (
        //eslint-disable-next-line react/no-array-index-key
        <Accordion key={i} trip={trip} />
      ))}
  </div>
);

export default createFragmentContainer(
  Multicity,
  graphql`
    fragment Multicity_booking on BookingMulticity {
      trips @relay(plural: true) {
        ...AccordionTripSummary_trip
      }
    }
  `,
);
