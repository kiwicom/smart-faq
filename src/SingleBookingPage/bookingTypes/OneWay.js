// @flow

import * as React from 'react';
import { createFragmentContainer, graphql } from 'react-relay';

import { simpleTracker } from '../../helpers/analytics/trackers';
import AccordionTitle from '../bookingItem/AccordionTitle';
import Accordion from '../bookingItem/Accordion';
import type { OneWay_booking } from './__generated__/OneWay_booking.graphql';

type Props = {|
  booking: OneWay_booking,
|};

const OneWay = ({ booking }: Props) => (
  <div>
    <AccordionTitle title="Outbound" />
    <div
      onKeyUp={null}
      role="button"
      tabIndex="-1"
      style={{ outline: 'none' }}
      onClick={simpleTracker('smartFAQBookingOverview', {
        action: 'openFlightCard',
      })}
    >
      <Accordion trip={booking.trip} />
    </div>
  </div>
);

export default createFragmentContainer(
  OneWay,
  graphql`
    fragment OneWay_booking on BookingOneWay {
      trip {
        ...AccordionTripSummary_trip
      }
    }
  `,
);
