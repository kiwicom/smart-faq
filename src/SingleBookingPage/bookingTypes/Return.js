// @flow

import * as React from 'react';
import { createFragmentContainer, graphql } from 'react-relay';

import { simpleTracker } from '../../helpers/analytics/trackers';
import TripTitle from '../bookingItem/AccordionTitle';
import Accordion from '../bookingItem/Accordion';
import type { Return_booking } from './__generated__/Return_booking.graphql';

type Props = {
  booking: Return_booking,
};

const Return = (props: Props) => (
  <div>
    <TripTitle title="Outbound" />
    <div
      onKeyUp={null}
      role="button"
      tabIndex="-1"
      style={{ outline: 'none' }}
      onClick={simpleTracker('smartFAQBookingOverview', {
        action: 'openFlightCard',
      })}
    >
      <Accordion trip={props.booking.outbound} />
    </div>
    <TripTitle title="Return" />
    <div
      onKeyUp={null}
      role="button"
      tabIndex="-1"
      style={{ outline: 'none' }}
      onClick={simpleTracker('smartFAQBookingOverview', {
        action: 'openFlightCard',
      })}
    >
      <Accordion trip={props.booking.inbound} />
    </div>
  </div>
);

export default createFragmentContainer(
  Return,
  graphql`
    fragment Return_booking on BookingReturn {
      outbound {
        ...AccordionTripSummary_trip
      }
      inbound {
        ...AccordionTripSummary_trip
      }
    }
  `,
);
