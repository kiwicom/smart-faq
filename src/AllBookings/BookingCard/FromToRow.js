// @flow

import idx from 'idx';
import * as React from 'react';
import { graphql, createFragmentContainer } from 'react-relay';
import {
  FlightReturn,
  FlightDirect,
  FlightMulticity,
} from '@kiwicom/orbit-components/lib/icons';

import bookingTypes from '../../common/booking/bookingTypes';
import type { FromToRow_arrival } from './__generated__/FromToRow_arrival.graphql';
import type { FromToRow_departure } from './__generated__/FromToRow_departure.graphql';

type Props = {
  type: string,
  departure: FromToRow_departure,
  arrival: FromToRow_arrival,
};

function renderFlightIcon(type: string) {
  switch (type) {
    case bookingTypes.ONE_WAY:
      return <FlightDirect size="medium" customColor="#bac7d5" />;
    case bookingTypes.RETURN:
      return <FlightReturn size="medium" customColor="#bac7d5" />;
    case bookingTypes.MULTICITY:
      return <FlightMulticity size="medium" customColor="#bac7d5" />;
  }
  return null;
}

const FromToRow = (props: Props) => {
  const origin = idx(props.arrival, _ => _.airport.city.name) || '';
  const destination = idx(props.departure, _ => _.airport.city.name) || '';
  const IATAOrigin = idx(props.arrival, _ => _.airport.locationId) || '';
  const IATADestination = idx(props.departure, _ => _.airport.locationId) || '';

  return (
    <div>
      <p className="flight">{`${origin} ${IATAOrigin}`}</p>
      <div className="arrowIcon">{renderFlightIcon(props.type)}</div>
      <p className="flight">{`${destination} ${IATADestination}`}</p>
    </div>
  );
};

export default createFragmentContainer(FromToRow, {
  arrival: graphql`
    fragment FromToRow_arrival on RouteStop {
      airport {
        locationId
        city {
          name
        }
      }
    }
  `,
  departure: graphql`
    fragment FromToRow_departure on RouteStop {
      airport {
        locationId
        city {
          name
        }
      }
    }
  `,
});
