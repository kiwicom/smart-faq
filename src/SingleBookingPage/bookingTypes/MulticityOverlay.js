// @flow

import * as React from "react";
import { createFragmentContainer, graphql } from "react-relay";

import Multicity from "./Multicity";
import type { MulticityOverlay_booking } from "./__generated__/MulticityOverlay_booking.graphql";

type Props = {|
  booking: MulticityOverlay_booking,
|};

type State = {|
  overlapping: boolean,
|};

class MulticityOverlay extends React.Component<Props, State> {
  state = {
    overlapping: true,
  };

  toggleOverlapping = () => {
    this.setState(prevState => ({ overlapping: !prevState.overlapping }));
  };

  render() {
    const { booking } = this.props;
    const { trips } = booking;

    return trips && trips.length > 2 ? (
      <div>
        <div
          className={
            this.state.overlapping
              ? "multicityWrapper"
              : "multicityWrapper multicityWrapperIsActive"
          }
        >
          <Multicity booking={booking} />
          <div
            className={
              this.state.overlapping
                ? "multicityOverlay"
                : "multicityOverlay multicityOverlayIsActive"
            }
          />
        </div>
        <button className="multicityButton" onClick={this.toggleOverlapping}>
          {this.state.overlapping ? "Show more trips" : "Show less trips"}
        </button>
        <style jsx>
          {`
            .multicityWrapper {
              position: relative;
              height: 215px;
              max-height: 215px;
              overflow: hidden;
              transition: all 1s ease-in-out;
            }
            .multicityWrapperIsActive {
              height: auto;
              max-height: 10000px; /* workaround for transition with height: auto */
              transition: all 1s ease-in-out;
            }
            .multicityOverlay {
              display: flex;
              justify-content: center;
              align-items: flex-end;
              position: absolute;
              top: 0;
              left: 0;
              width: 100%;
              height: inherit;
              background: linear-gradient(
                to bottom,
                rgba(225, 0, 0, 0) 50%,
                rgba(225, 0, 0, 0) 30%,
                rgba(245, 247, 249, 0.59) 30%,
                rgb(245, 247, 249) 100%
              );
            }
            .multicityOverlayIsActive {
              animation: overlay 1s ease-in-out forwards;
            }
            .multicityButton {
              width: 100%;
              background: transparent;
              border: none;
              font-weight: bold;
              text-decoration: underline;
              font-size: 0.9rem;
              cursor: pointer;
            }
            .multicityButton:focus {
              outline: 0;
            }
            @keyframes overlay {
              0% {
                opacity: 1;
              }
              100% {
                opacity: 0;
                z-index: -1;
              }
            }
          `}
        </style>
      </div>
    ) : (
      <Multicity booking={booking} />
    );
  }
}

export default createFragmentContainer(
  MulticityOverlay,
  graphql`
    fragment MulticityOverlay_booking on BookingMulticity {
      trips {
        duration
      }
      ...Multicity_booking
    }
  `,
);
