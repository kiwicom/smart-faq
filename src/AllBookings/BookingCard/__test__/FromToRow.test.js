// @flow

import * as React from 'react';
import { shallow, render } from 'enzyme';

import bookingTypes from '../../../common/booking/bookingTypes';
import { FromToRow } from '../FromToRow';

const mockRefType: any = null;

const arrival = {
  airport: {
    city: {
      name: 'Barcelona',
    },
    locationId: 'BCN',
  },
  $refType: mockRefType,
};
const departure = {
  airport: {
    city: { name: 'Prague' },
    locationId: 'PRG',
  },
  $refType: mockRefType,
};

const fromToRowComponent = (
  <FromToRow
    type={bookingTypes.MULTICITY}
    arrival={arrival}
    departure={departure}
  />
);

describe('FromToRow', () => {
  it('should render correctly', () => {
    expect(shallow(fromToRowComponent)).toMatchSnapshot();
  });

  it('should render Multicity icon', () => {
    const component = shallow(fromToRowComponent);
    expect(component.find('FlightMulticity').exists()).toBeTruthy();
  });

  it('should show IATA codes', () => {
    const wrapper = render(fromToRowComponent);
    expect(wrapper.text()).toContain('PRG');
    expect(wrapper.text()).toContain('BCN');
  });

  it('should show city names', () => {
    const wrapper = render(fromToRowComponent);
    expect(wrapper.text()).toContain('Prague');
    expect(wrapper.text()).toContain('Barcelona');
  });
});
