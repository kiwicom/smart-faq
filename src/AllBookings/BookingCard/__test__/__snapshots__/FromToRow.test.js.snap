// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FromToRow should render correctly 1`] = `
<div>
  <p
    className="flight"
  >
    Barcelona BCN
  </p>
  <div
    className="arrowIcon"
  >
    <FlightMulticity
      customColor="#bac7d5"
      size="medium"
    />
  </div>
  <p
    className="flight"
  >
    Prague PRG
  </p>
</div>
`;
